packages/headless/src/controllers/tab/headless-tab.ts

```diff
@@ -80,7 +80,7 @@ export function buildTab(
     'buildTab'
   );
 
-  if (initialState.isActive) {
+  if (initialState.isActive && engine.state.advancedSearchQueries.cq === null) {
```

Comment:
This is the most important part of this PR, but it has a flaw: if somebody intends to change which tab is currently active by creating a new controller whose `isActive` is true, it won't work anymore. Personally, I don't think that's much of a problem since in order for that to happen, a developer would need to be attempting to manage tabs themselves instead of reading the intended `state.isActive` and calling the intended `select`.

Reply:
Agreed, there should only ever be one Tab controller with `isActive`. If a dev wants to make another tab active, they would have to remove the `isActive` on the other instance.

As for the condition, I think we should have this inside the reducer itself, and have the controller dispatch a special `registerAdvancedSearchQueries` or `registerConstantQuery` action.